### PR TITLE
PLT-7924: Added joining/leaving channels to load test

### PR DESCRIPTION
**Pull request by MusikPolice**

### CODE REVIEW

File: loadtest/tests.go [hunk 1]
```diff
@@ -151,6 +151,72 @@ func actionPost(c *EntityConfig) {
...
+		}
+	}
+
+	// get channels the user is already a member of on that team
+	var channelId *string
+	if channelMembers, resp := c.Client.GetChannelMembersForUser(userId, teamId, ""); resp.Error != nil {
+		cmdlog.Errorf("Unable to get channel members for user. User: %v, User: %v, Error: %v", userId, c.UserData.Username, resp.Error.Error())
+		return
+	} else {
+		// find the first channel on the team that the user isn't already a member of, and use it for the test
```
Comment (MusikPolice) [inline loadtest/tests.go:196]: This is more or less what the [redux selector](https://github.com/mattermost/mattermost-redux/blob/master/src/selectors/entities/channels.js#L216) that powers the "More" link in the LHS does. Golang doesn't have a `filter` function, so I had to use loops instead, but it's fairly realistic.

File: loadtest/tests.go [hunk 2]
```diff
@@ -151,6 +151,72 @@ func actionPost(c *EntityConfig) {
 	}
 }
 
+func actionJoinLeaveChannel(c *EntityConfig) {
+	// figure out my user id
+	var userId string
+	if user, resp := c.Client.GetUserByEmail(c.UserData.Email, ""); resp.Error != nil {
+		cmdlog.Errorf("Unable to get user by email. User: %v, Error: %v", c.UserData.Username, resp.Error.Error())
```
Comment (ccbrown) [inline loadtest/tests.go:164]: Should there be a `return` here?

File: loadtest/tests.go [hunk 3]
```diff
@@ -151,6 +151,72 @@ func actionPost(c *EntityConfig) {
...
+		return
+	} else {
+		channelIdsInTeam = make([]string, 0)
+		for _, channel := range channels {
+			channelIdsInTeam = append(channelIdsInTeam, channel.Id)
+		}
+	}
+
+	// get channels the user is already a member of on that team
+	var channelId *string
```
Comment (ccbrown) [inline loadtest/tests.go]: Simpler and more idiomatic to just use `string` when the empty string isn't a valid value.

**Review by ccbrown — APPROVED** [review]
Some very minor things, but lgtm.

File: loadtest/tests.go (hunk 2 above)
Comment (MusikPolice) [reply loadtest/tests.go:164]: Fixed in https://github.com/mattermost/mattermost-load-test/pull/52/commits/c473fabd9aaf6ebbf63b60ac27eb1e0f47d9dcc0

File: loadtest/tests.go [hunk 4]
```diff
@@ -344,10 +412,14 @@ var standardUserEntity UserEntity = UserEntity{
 			Item:   actionDisconnectWebsocket,
 			Weight: 2,
 		},
+		{
```
Comment (crspeller) [inline loadtest/tests.go:421]: I doubt this is a realistic weight. Can you remove it from here and we can add it back later with a realistic weight?

**Review by crspeller — CHANGES_REQUESTED** [review]

**Comment by MusikPolice** [comment]
@crspeller reminder that we need real weights for this test so that it can be merged. I'm happy to do the work to determine them if you tell me how you go about it.

**Comment by MusikPolice** [comment]
@crspeller Can you take a look at the weighting on this so I can get it merged?

**Comment by MusikPolice** [comment]
@crspeller Can you take a look at the weights on this so that I can merge it?

**Pull request closed without merging**